cmd/kseal/cmd/encrypt/encrypt.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2022 - Bitnami <containers@bitnami.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package encrypt
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// NewCmdEncrypt creates a command object for the "encrypt" action.
+func NewCmdEncrypt() *cobra.Command {
+	encryptCmd := &cobra.Command{
+		Use:   "encrypt",
+		Short: "Encrypt a secret",
+		Long: `Encrypt a secret using a public key.
+
+Examples:
```

Comment:
I think the main UX issue in `kubeseal` is the fact that we relied too much on people "feeding" in a k8s Secret as input to Seal.

I still think that's a very useful feature, but the new command should offer a more intuitive structure for creating and updating SealedSecret files.

One of the common pitfalls in `kubeseal` was that it was expecting data on `stdin` by default. That behavior fits very well in the unix philosophy but unfortunately a lot of users are not proficient in the unix way. 

Another source of confusion was the distinction between encrypting values and producing SealedSecret files (which contain encrypted items). Some people care about one aspect, some care about the other (hence the "--raw" API).

Proposal:

1. add a `create` command that mirrors closely the behaviour of `kubectl create secret`:

```
$ kseal create foo -n myns --from-literal=item1=value
$ kseal create foo -n myns --from-file=item1=file.txt --from-literal=item2=value
$ kseal create foo -n myns --from-env-file=.env
$ kseal create foo -n myns --from-literal=item1=value
$ kseal create foo -n myns --from-literal=item1=value --type=my-secret-type
$ kseal create foo --scope cluster-wide --from-literal=item1=value
```

2. Add the ability to create a sealed secrets from an existing k8s secret:

```
$ kseal create --from-secret=existing-secret.yaml
```

3. Consistent command for updating existing secrets (aka `--merge-into`):

```
$ kseal update sealed-secret.yaml --from-literal=item1=value
```

4. Ability to produce individual encrypted data items (tentative name scheme)

```
$ kseal data encrypt --sealed-secrets-context=sealed-secrets.yaml --from-file=foo.txt 
$ kseal data encrypt --sealed-secrets-context=sealed-secrets.yaml --from-literal=bar
$ kseal data encrypt --name foo -n myns --scope namespace-wide --from-literal=baz
```

Reply:
Brilliant! I guess this would be also a replacement for the `--raw` mode, right?

Reply:
yes